lib/ansible/cli/doc.py

```diff
@@ -35,9 +35,22 @@
 display = Display()
 
 
+def _json_fix_keys(obj):
+    '''Make sure that the data does not contain mappings with non-string keys which break JSON encoding.'''
```

Comment:
do we have any 'valid' keys that would cause this? at this point i think we SHOULD error out and inform user that the plugin author did something wrong.

Reply:
Depends on how you define 'valid'. I guess validate-modules could be more strict about what is accepted for module return value `sample`s. That's where the bad values were hiding in the cases I found.

Reply:
I'll create a PR for stronger validation (both validate-modules and ansible-doc sanity test) this weekend. That would also solve this.

Reply:
but sample is the key, it could have any value, and there I do expect such things to occur, but not on the keys themselves.

Reply:
Sorry for being imprecise. It is not the key `sample`, but what is the value for `sample`. Often this is not text, but also something that the YAML parser parses. And for example for  `ec2_vpc_vpn` in stable-2.9 (already fixed in the collection and backport exists which fixes this, so it won't "work" forever), the `sample` value for return value `vgw_telemetry` contains a structure which ends up being parsed by the YAML parser as something which contains a `dict` where one key is not a string.

Reply:
PRs for more sanity tests are #69287 and #69288.